holonote/annotate/annotator.py

```diff
@@ -195,72 +190,127 @@ def selected_index(self):
         "Convenience property returning a single selected index (the first one) or None"
         return self.selected_indices[0] if len(self.selected_indices) > 0 else None
 
-
-    def _get_range_indices_by_position(self, dim1_pos, dim2_pos=None):
-        ranges = self.annotation_table._region_df[self.annotation_table._region_df['region_type']=='Range']
-        mask = ranges['value'].apply(lambda tpl: tpl[0] <= dim1_pos < tpl[1])
-        if dim2_pos is not None:
-            mask_dim2 = ranges['value'].apply(lambda tpl: tpl[2] <= dim2_pos < tpl[3])
-            mask = mask & mask_dim2
-        if not len(mask):
+    def _get_range_indices_by_position(self, **inputs) -> list[Any]:
+        df = self.annotation_table._region_df
+        ranges = df[df['region']=='range']
+        if ranges.empty:
             return []
-        return list(ranges[mask]['_id'])
 
+        for i, (k, v) in enumerate(inputs.items()):
+            dim = ranges[ranges["dim"] == k]
+            mask = dim["value"].apply(lambda d: d[0] <= v < d[1])  # noqa: B023
+            if i == 0:
+                ids = set(dim[mask]._id)
+            else:
+                ids &= set(dim[mask]._id)
+        return list(ids)
 
-    def get_indices_by_position(self, dim1_pos, dim2_pos=None):
+    def get_indices_by_position(self, **inputs) -> list[Any]:
         "Return primary key values matching given position in data space"
         # Lots TODO! 2 Dimensions, different annotation types etc.
-        range_matches = self._get_range_indices_by_position(dim1_pos, dim2_pos)
+        range_matches = self._get_range_indices_by_position(**inputs)
         event_matches = [] # TODO: Needs hit testing or similar for point events
         return range_matches + event_matches
 
     @property
     def region(self):
         return self._region
 
+    def set_regions(self, **items):
+        self._set_regions(**items)
+
+    def _set_regions(self, **items):
```

Comment:
Isn't this redundant with `set_regions`?

Reply:
This is loosely following the structure of the existing code. 

In the `Annotator` class, `set_regions` will also call `self.refresh`. This will give problems when running some of the holoviews code where it will keep on refreshing. 

I have made a small comment about it here:
https://github.com/holoviz/holonote/blob/decouple_plot_annotator/holonote/annotate/annotator.py/#L548